src/services/device.service.ts

```diff
@@ -0,0 +1,31 @@
+import { db } from '../database';
+
+export interface Device {
+  userId: number;
+  deviceToken: string;
+}
+
+interface InsertDeviceResult {
+  userId: number;
+  deviceToken: string;
+}
+
+interface DeviceCreated {
+  userId: number;
+  deviceToken: string;
+}
+
+const addDevice = async (device: Device): Promise<DeviceCreated> => {
+  const result = await db.sql<InsertDeviceResult>('insert-devices', {
+    user_id: device.userId,
+    device_token: device.deviceToken,
+  });
+  return {
+    userId: result.rows[0].userId,
+    deviceToken: result.rows[0].deviceToken,
+  };
+};
```

Comment:
I apologize for suggesting a nearly complete rewrite of this file, but not `RETURNING` anything from the `INSERT` allows us to simplify dramatically: we no longer need interfaces for what the database returns or what the function returns, and we don't need to translate that response from snake_case to camelCase:

```suggestion
const addDevice = async (device: Device): Promise<void> => {
  await db.sql('insert-devices', {
    user_id: device.userId,
    device_token: device.deviceToken,
  });
};
```

This returns a `Promise<void>`, which means it either succeeds or errors, but does not contain any additional information.